### ISSUE TITLE
django-celery-beat does not support nowfun in schedules

### ISSUE THREAD

**Issue by liberza** [celery/django-celery-beat#49]
See http://docs.celeryproject.org/en/latest/reference/celery.schedules.html

Even when using CrontabSchedule.from_schedule() with a schedule that uses nowfun, django-celery-beat does not store nowfun or make use of it.

```
from django_celery_beat.models import CrontabSchedule
from celery.schedules import crontab
nowfun = lambda: datetime.now(pytz.timezone("US/Eastern"))
crontab_schedule = crontab(minute=minute,
    hour=1,
    day_of_month="*",
    month_of_year="*",
    day_of_week="*",
    nowfun=nowfun)
db_crontab_schedule = CrontabSchedule.from_schedule(crontab_schedule)
db_crontab_schedule.save()
```

The above does not end up saving or using nowfun.

**Comment by mapleflow** [comment]
in `django-celery-beat/model.py`, it does not support nowfun, i think you can extend `CrontabSchedule` to solve your problem.

**Comment by liberza** [comment]
Yes, I'm exploring options for extending it. Could pickle the lambda expression and store the string in CrontabSchedule, but that's pretty ugly... not sure how else to do it while making it compatible with already-existing schedules containing a nowfun, though.

**Comment by Jorl17** [comment]
Hi.

Is there any predicted support for this feature? It is quite valuable to set events that are timezone aware.

Thanks

**Comment by thedrow** [comment]
I believe https://github.com/celery/django-celery-beat/commit/b6c0581756a7e47e356a5ca64c674142402f9237 fixes this issue and it is available in 1.1.0.
Please let me know if I'm wrong and I'll reopen this issue.

**Comment by Jorl17** [comment]
We still get
`TypeError: can't pickle function objects` when trying to use
`"schedule": crontab(hour=14, minute=38, nowfun=zonenowfun)`

with `zonenowfun = lambda: datetime.datetime.now(one_zone)`

and `1.1.0`

Any clues as to what is the problem?

**Comment by thedrow** [comment]
You cann't pickle lambdas. Use a normal function.

**Comment by Jorl17** [comment]
A normal function, in this particular context, has the same issue because it is being created within a for loop which iterates over different timezones (we create a different task for each timezone).

My main gripe with this issue is that this code was working with djcelery and it is pretty much taken from several online tutorials using celery and djcelery. When I upgraded a relatively legacy project to new celery versions (which have deprecated djcelery), this completely broke down, and there seems to be no way to do the same thing. The Celery documentation also does not state anything regarding limitations with nowfun.

Is there no planned fix for this? In my project we have worked around the issue, but it might hit other people. As an example, [this thread](https://stackoverflow.com/questions/21827290/celery-beat-different-time-zone-per-task) often comes up in search results for celery and timezones.
